README.md

```diff
@@ -622,6 +622,52 @@ const storage = createStorage({
 - `storage`: The name of the table to read from. It defaults to `storage`.
 - `boostCache`: Whether to enable cached queries: see [docs](https://planetscale.com/docs/concepts/query-caching-with-planetscale-boost#using-cached-queries-in-your-application).
 
+### `azure-key-vault`
+
+⚠️ This driver is currently not compatible with edge workers like Cloudflare Workers or Vercel Edge Functions.
+
+Store data in a Azure Key Vault [secrets](https://docs.microsoft.com/en-us/azure/key-vault/secrets/about-secrets).
+
+This driver stores KV information in Azure Key Vault secrets by using the key as secret id and the value as secret content.
+Please be aware that key vault secrets don't have the fastest access time and are not designed for high throughput. You also have to disable purge protection for your key vault to be able to delete secrets. This implementation deletes and purges a secret when it is deleted to avoid conflicts with soft delete.
+
+To use it, you will need to install `@azure/keyvault-secrets` and `@azure/identity` in your project:
+
+```json
+{
+  "dependencies": {
+    "@azure/keyvault-secrets": "^4.6.0",
+    "@azure/identity": "^3.1.3"
+  }
+}
+```
+
+Usage:
+
+```js
+import { createStorage } from "unstorage";
+import azureKeyVault from "unstorage/drivers/azure-key-vault";
+
+const storage = createStorage({
+    driver: azureKeyVault({
+      vaultName: "testunstoragevault",
+  }),
+});
+```
+
+**Authentication:**
+
+The driver supports the following authentication methods:
+
+- **`DefaultAzureCredential`**: This is the recommended way to authenticate. It will use managed identity or environment variables to authenticate the request. It will also work in a local environment by trying to use Azure CLI or Azure PowerShell to authenticate. <br>
+⚠️ Make sure that your Managed Identity or personal account has either the `Key Vault Secrets Officer` RBAC role assigned or being member of an access policy that grants `Get`, `List`, `Set`, `Delete` and `Purge` secret permissions.
```

Comment:
```suggestion
- **`DefaultAzureCredential`**: This is the recommended way to authenticate. It will use managed identity or environment variables to authenticate the request. It will also work in a local environment by trying to use Azure CLI or Azure PowerShell to authenticate.

   ⚠️ Make sure that your Managed Identity or personal account has either the `Key Vault Secrets Officer` RBAC role assigned or is a member of an access policy that grants `Get`, `List`, `Set`, `Delete` and `Purge` secret permissions.
```